pymc/sampling/forward.py

```diff
@@ -497,22 +499,85 @@ def sample_posterior_predictive(
 
     Examples
     --------
-    Thin a sampled inferencedata by keeping 1 out of every 5 draws
-    before passing it to sample_posterior_predictive
+    Sample posterior predictive samples any variable included in `var_names`,
+    conditioned on the values of other random variables found in the trace.
+    To ensure the samples are internally consistent, any random variable that depends
+    on another random variable that is being sampled is also sampled.
+    The final list of variables being sampled is shown in the log output.
+
+    Notice that if a random variable has no dependency on other random variables,
```

Comment:
I'd put this in a :: warning box for the Sphinx api docs, and strengthen the language. Suggested:

**Important**: If a random variable does not depend on any other random variables, it will simply be sampled from itself -- that is, from the prior distributed defined in the model. **This function will _never_ result in a copy of a random variable's posterior distribution being returned**

Reply:
That's a great idea!

Reply:
I added below, instead of replacing the text. WDYT?

Reply:
Yes I like how you did it